src/main/java/rx/Observable.java

```diff
@@ -1258,6 +1258,7 @@ public Completable toCompletable() {
         final static Observable<Object> INSTANCE = create(new OnSubscribe<Object>() {
             @Override
             public void call(Subscriber<? super Object> subscriber) {
+                subscriber.setProducer(ProducerHolder.INSTANCE);
```

Comment:
Setting the producer is optional in 1.x and many other operators don't do it either; one has to work around it. Why add it just here?